pennylane/math/utils.py

```diff
@@ -138,7 +138,9 @@ def cast_like(tensor1, tensor2):
     >>> cast_like(x, y)
     tensor([1., 2.])
     """
-    if not is_abstract(tensor2):
+    if isinstance(tensor2, ArrayBox):
+        dtype = ar.to_numpy(tensor2._value).dtype.type  # pylint: disable=protected-access

```

Comment:
does this not break anything? why did this need to be added?

Reply:
It was breaking something, but now I'm forgetting what.